lib/semian/circuit_breaker.rb

```diff
@@ -51,14 +51,13 @@ def mark_failed(error)
 
     def mark_success
       return unless half_open?
-      @success_count += 1
+      push_time(@successes, @success_count_threshold)
```

Comment:
If the outage is long then the circuit will reopen immediately on the first error, so no even for long outages it's fine.